Sources/Subscription/SubscriptionManager.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  SubscriptionManager.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+public protocol SubscriptionManaging {
+
+    var accountManager: AccountManaging { get }
+    var subscriptionService: SubscriptionService { get }
+    var authService: AuthService { get }
+    @available(macOS 12.0, iOS 15.0, *)
+    func getStorePurchaseManager() -> StorePurchaseManaging
```

Comment:
yes, this is a way to support iOS 14, `let storePurchaseManager: StorePurchaseManaging?` is optional just because StorePurchaseManagER can't be used in iOS14, but using it across the optional as optional is a huge pain, this allows the app to have a non optional StorePurchaseManagER where needed. I can change name but that seems explicative enough

Reply:
Not sure if we have it as a pattern but what do you think about making it like it is done in https://github.com/duckduckgo/macos-browser/blob/15ce1f646867364c23f118c75dc3adaa1b82fbee/LocalPackages/SubscriptionUI/Sources/SubscriptionUI/DebugMenu/SubscriptionDebugMenu.swift#L41

Reply:
What pattern are you referring to? A function like getXXX() that returns an instance of XXX?
The code you linked has a different purpose, serving the instance and initialising it if needed, this is only an accessor and the init is, on purpose, handled somewhere else.
What is it that you don't like? getXXX and setXXX is how getters and setters are implemented under the hood.

Reply:
I meant to implement it as a var and ditch the `get` prefix, as in Swift (and even before that) there was no pattern to name getters like that. In the linked example there is additional need to hide the type which has restricted availability. In general I suggest `get` less var with non-optional type and `_` prefixed storage var being optional.

Reply:
the iOS app contains 168 getXXX() functions, I think this is only an aesthetic preference, but I don't want to drag this out for too long.
I renamed `getStorePurchaseManager()` to `storePurchaseManager()`, I can't create a var because of the dummy class `SubscriptionManageriOS14` that needs to declare the accessor `@available(iOS 15, *)` and can't do it on a var